pennylane/templates/embeddings.py

```diff
@@ -0,0 +1,207 @@
+
+    .. note::
+
+        AmplitudeEmbedding uses PennyLane's :func:`QubitStateVector()` and only works in conjunction with
+        devices that implement this function.
+
+    Args:
+        features (array): Input array of shape ``(N, )``, where N is the number of input features to embed
+        n_qubits (int): Number of qubits in the circuit
+    """
+
+    if 2**n_qubits != len(features):
+        raise ValueError("AmplitudeEmbedding requires a feature vector of size 2**n_qubits which is {}, "
+                         "got {}.".format(2**n_qubits, len(features)))
+
+    QubitStateVector(features, wires=range(n_qubits))
+
+
+def SqueezingEmbedding(features, n_wires, execution='amplitude', c=0.1):
+    r"""
+    Encodes the entries of ``features`` into the squeezing phases :math:`\phi` or amplitudes :math:`r` of the modes of
+    a continuous-variable quantum state.
+
+    The mathematical definition of the squeezing gate is given by the operator
+
+    ..math::
+
+        S(z) = \exp\left(\frac{r}{2}\left(e^{-i\phi}\a^2 -e^{i\phi}{\ad}^{2} \right) \right),
+
+    where :math:`\a` and :math:`\ad` are the bosonic creation and annihilation operators.
+
+    ``features`` has to be an array of ``n_wires`` floats.
+
+    Args:
+        features (array): Binary sequence to encode
+        n_wires (int): Number of qubits in the circuit
+        execution (str): ``'phase'`` encodes the input into the phase of single-mode squeezing, while
+            ``'amplitude'`` uses the amplitude.
+        c (float): parameter setting the value of the phase of all squeezing gates if ``execution='amplitude'``, or the
+            amplitude of all squeezing gates if ``execution='phase'`` to a constant.
+    """
+
+    if n_wires != len(features):
+        raise ValueError("SqueezingEmbedding requires a feature vector of size n_wires which is {}"
+                         ", got {}.".format(n_wires, len(features)))
+
+    for i in range(n_wires):
+        if execution == 'amplitude':
+            Squeezing(features[i], c, wires=i)
+        elif execution == 'phase':
+            Squeezing(c, features[i], wires=i)
+        else:
+            raise ValueError("Execution strategy {} not known. Has to be `phase` or `amplitude`.".format(execution))
+
+
+def DisplacementEmbedding(features, n_wires, execution='amplitude', c=0.1):
+    r"""
+    Encodes the entries of ``features`` into the displacement phases :math:`\phi` or amplitudes :math:`r` of the modes of
+    a continuous-variable quantum state.
+
+    The mathematical definition of the displacement gate is given by the operator
+
+    ..math::
+            D(\alpha) = \exp(r (e^{i\phi}\ad -e^{-i\phi}\a)),
+
+    where :math:`\a` and :math:`\ad` are the bosonic creation and annihilation operators.
+
+    ``features`` has to be an array of ``n_wires`` floats.
+
+    Args:
+        features (array): Binary sequence to encode
+        n_wires (int): Number of qubits in the circuit
```

Comment:
qubits -> wires

Reply:
I would like the reference to qubits here, since the embedding works with qubits. So n_wires IS the number of qubits in the circuit, even though we standarise the keyword argument.

Reply:
:+1: